core/score_sync.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2022 Board of Trustees of the University of Illinois.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	"application/core/model"
+	"log"
+	"os"
+	"strconv"
+)
+
+// SyncAmgUUIDForScore fetches the AmgUUID from Core BB and populates the external_user_id field
+func (app *Application) SyncAmgUUIDForScore(score *model.Score, externalIDs map[string]string) error {
+	// Get Mastodon ID from externalIDs if provided, otherwise use the one already in score
+	mastodonID := score.ExternalProfileID
+	if externalIDs != nil {
+		if mastodonIDFromMap, ok := externalIDs["mastodon_id"]; ok && mastodonIDFromMap != "" {
+			mastodonID = mastodonIDFromMap
+		}
+	}
+
+	if mastodonID == "" {
+		log.Printf("SyncAmgUUIDForScore: no mastodon_id provided for score, skipping AmgUUID sync")
+		return nil
+	}
+
+	// Look up the AmgUUID from Core BB
+	accountCriteria := map[string]interface{}{
+		"identifiers": map[string]interface{}{
+			"$elemMatch": map[string]interface{}{
+				"code":       "mastodon_id",
+				"identifier": mastodonID,
+			},
+		},
+	}
```

Comment:
While this doesn't depend on mongo packages explicitly, this syntax is still strictly tied to mongo, so it should also go in the `storage` package (in driven/storage).